spring/src/test/java/org/axonframework/spring/authorization/AuthorizationMessageDispatchInterceptorTest.java

```diff
@@ -0,0 +1,88 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.spring.authorization;
+
+import org.axonframework.messaging.correlation.SimpleCorrelationDataProvider;
+import org.axonframework.messaging.interceptors.CorrelationDataInterceptor;
+import org.axonframework.test.aggregate.AggregateTestFixture;
+import org.axonframework.test.aggregate.FixtureConfiguration;
+import org.axonframework.test.matchers.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.UUID;
+
+import static org.hamcrest.core.StringStartsWith.startsWith;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {AuthorizationMessageDispatchInterceptor.class, MessageAuthorizationInterceptor.class})
+class AuthorizationMessageDispatchInterceptorTest {
+
+    private FixtureConfiguration<TestAggregate> fixture;
+
+    @BeforeEach
+    public void setUp() {
+        fixture = new AggregateTestFixture<>(TestAggregate.class);
+    }
+
+    @Test
+    @WithMockUser(username = "admin", authorities = {"ROLE_aggregate.create"})
+    public void shouldAuthorizeAndPropagateUsername() {
+        UUID aggregateId = UUID.randomUUID();
+        fixture.registerCommandDispatchInterceptor(new AuthorizationMessageDispatchInterceptor<>())
+               .registerCommandHandlerInterceptor(new MessageAuthorizationInterceptor<>())
+               .registerCommandHandlerInterceptor(new CorrelationDataInterceptor<>(new SimpleCorrelationDataProvider(
+                       "username")))
+               .given()
+               .when(new CreateAggregateCommand(aggregateId))
+               .expectSuccessfulHandlerExecution()
+               .expectResultMessageMatching(Matchers.matches(message -> ((User) message.getMetaData()
+                                                                                       .get("username")).getUsername()
+                                                                                                        .equals("admin")));
+    }
+
+    @Test
+    public void shouldNotAuthorizeOnNoAuthentication() {
+        UUID aggregateId = UUID.randomUUID();
+        fixture.registerCommandDispatchInterceptor(new AuthorizationMessageDispatchInterceptor<>())
+               .registerCommandHandlerInterceptor(new MessageAuthorizationInterceptor<>())
+               .registerCommandHandlerInterceptor(new CorrelationDataInterceptor<>(new SimpleCorrelationDataProvider(
+                       "username")))
+               .given()
+               .when(new CreateAggregateCommand(aggregateId))
```

Comment:
Curious: why not validate `.expectException(UnauthorizedMessageException.class)` in this test?